### ISSUE TITLE
"Failed getting PPFT value" after resuming from standby

### ISSUE THREAD

**Issue by xnoreq** [EionRobb/skype4pidgin#603]
Pidgin 2.12.0 (libpurple 2.12.0)
skype4pidgin 1.4 

After coming out of standby, pidgin fails to reconnect to skype: "Failed getting PPFT value".

Also, what does the empty checkbox in advanced account settings do?

![grafik](https://user-images.githubusercontent.com/4028859/38162190-40b95230-34dd-11e8-82d7-bb32790a4e80.png)

**Comment by EionRobb** [comment]
Have you followed the flowchart in the readme?

**Comment by xnoreq** [comment]
Yes, I was guessing that the empty checkbox is the "alt login" checkbox, but checking it didn't help anything either.
So since I cannot build the latest git version I had to report a bug.

I also did the following experiment:
1) connect in pidgin
2) disable the network adapter
Now, interestingly, the skype plugin doesn't seem to notice that it doesn't have a connection anymore. Maybe it does only after a couple of minutes (when some timeout is reached)?
3) disconnect in pidgin
4) try to connect again (with the network adapter still disabled): "Failed getting PPFT value"

Even after enabling the network adapter again, it can take a few seconds until the network connection is re-established. During that time, any connection attempts in pidgin will of course also fail.

Maybe after resuming from standby the skype plugin tries to connect too soon (before the network connection is re-established) and then simply gives up?

**Comment by xnoreq** [comment]
I have edited the first post to clarify that only the automatic reconnection fails. I can still reconnect manually, but it's annoying if I have to check the connection all the time and reconnect manually.

**Comment by bolera** [comment]
I would like to re-affirm that this is still an issue with exactly the same scenario as posted by the original reporter: it happens after a standby. If I restart the PC it normally does not happen. Not sure about a cold start (I almost never do that). I think skypeweb simply gives up too fast or tries to connect too early. So, the solution would be to wait a bit for the first connect or retry a bit more "conservative". I think a one-minute delay would already work. If you look at #474 you see comments about patches to skypeweb.dll doing that. Well, they don't change the connection attempt itself, but the error code, so that it isn't fatal anymore and reconnection attempts can happen. The "alternative auth" isn't going to work according to the flow chart, because it is only for two-factor authentication. (Btw, I had a hard time finding that mysterious "readme" as it isn't at https://github.com/EionRobb/skype4pidgin nor in the source code that I downloaded. I finally found it this way: Wiki tab ->https://wiki.bitlbee.org/HowtoSkypeWeb -> https://github.com/EionRobb/skype4pidgin/tree/master/skypeweb#skypeweb-plugin-for-pidgin). If I understand correctly, the alternative auth can only be used in conjunction with bitlbee, anyway?
It would be nice if that timeout period (or whatever you want to call it) could be extended if the first attempt fails or any other suitable solution that makes it reconnect. #474 also mentions that Pidgin tries to reconnect by itself, anyway, so maybe changing the "fatality" of the error would suffice, indeed. Thanks!

**Comment by EionRobb** [comment]
The alternate auth isn't just for people with a 2fa, it'll work for everyone.  I should really make it the default and get rid of the older login mech.

Try using that setting and see how you get on.

**Comment by bolera** [comment]
Ok, thanks. I just followed the flow chart ;-) So, how do I enable "alternative auth"? Is it that checkmark under Modify Account/Advanced that doesn't have a label? If so, I did now. Will report back. May take some days.

**Comment by bolera** [comment]
Hi Robb, *if* that checkmark without a label sets "alternative auth" it doesn't make a difference. It's been the same today as with all other wakeups from hibernation in the past. I've verified it's still checked. So, is that the checkmark that sets it or is it elsewhere?

**Comment by EionRobb** [comment]
@bolera 
![image](https://user-images.githubusercontent.com/1063865/53946289-5316a100-4128-11e9-9ee8-3a74ae2b51dc.png)
If you haven't got that option then you're on an older version of the plugin and should consider upgrading

**Comment by bolera** [comment]
I see I'm also missing the Protocol Icon option.
I think I'm up-to-date, there haven't been updates for a while. Which file would that be?
LibPurple 2.13.0.0 (it came with Pidgin 2.13.0)
libskypeweb.dll has no version information, it's dated June 6, 2018, so I believe it's 1.5. I also have the 1.4 dll available which is smaller and a year older.

**Comment by bolera** [comment]
Editing doesn't work here for me, so I'm adding. I compared with the content of pidgin-skypeweb-installer.exe 1.5 from here: https://github.com/EionRobb/skype4pidgin/releases (usually I just download the dll). And I copied it over. It's identical and no change on that property sheet.

**Comment by bolera** [comment]
I had kept a libskypeweb.14.dll in the plugins directory. Just in case. It seems that Pidgin scans all dlls in it and uses the first (or last, not sure) API it finds. So, it was obviously using the libskypeweb.14.dll and not the libskypeweb.dll in it. I now have the new login option (but not the Protocol icon option) and checked it. Let's see what happens tomorrow morning.
Isn't it a security risk that Pidgin just scans the whole directory for dlls and inserts them as plugins?

**Comment by bolera** [comment]
Hello Robb, I can confirm that changing to the "alternative login method" fixes this. It connects now just fine after a hibernate. Thanks!

**Comment by brianjmurrell** [comment]
As of Mon Sep  9 17:39:00  2019 UTC, without making any changes to my configuration for this plugin I am now getting this error yet again.

I have logged in from an incognito window and that doesn't fix it.

_Alternative login method_ reports **Error getting BinarySecurityToken**.

I'm using git commit 5d29285 which appears to be the latest version.

**Comment by grzegorz-janoszka** [comment]
Hi,
I have the same issue with PPFT value after waking up the computer. When I switch to the alternate login method I see: "No handler found for resource (code 40499)" and it doesn't connect at all.

**Comment by grzegorz-janoszka** [comment]
I think the issue is when you have an USB network card. My laptop has USB 3.0 docking station and the network comes via it. So far every time I wake my laptop I have to manually enable the protocol. It is very annoying and sometimes I forget it and I am offline for the whole day.

What can we do for this plugin to handle better sleep/wake events?

**Comment by bsdoss** [comment]
i use skypeweb 1.7 and suffer from "Failed getting PPFT value, please try logging in via browser first", even i log in via browser first.

**Comment by chkp** [comment]
> i use skypeweb 1.7 and suffer from "Failed getting PPFT value, please try logging in via browser first", even i log in via browser first.

Same for me, logging in from browser works fine - there I also signed out and in again, but issue remains.
Checked sign in activity, I only see successful logins.,  
Unfortunately alternative login method results in "bad username or password". 
But despite the issue, thanks a lot for all your work on skype4pidgin!

**Comment by chkp** [comment]
It works again for me. instead of using the v1.7 library from the download-page (releases), I cloned the source and built a .deb. The library is much smaller than the downloaded one but it works for now...  crossing fingers :-)
Btw, the alternative login method is still failing.
